### DISCUSSION TITLE
Function code explanation "void rp2040pwmIrq(void)".

### DISCUSSION THREAD

**Discussion by alessandro-satanassi** [TeamOpenFIRE/OpenFIRE-Firmware#15] [General]
What is the purpose of the code "pwm_hw->intr = 0xff;" at line 922 of the SamcoEnhanced.ino file? Why is there such a code?

**Comment by SeongGino** [comment]
If you'd look at what it's attached to, you'd notice that it's part of the hardware timer to flag when the camera is ready to be read.

Is there any reason you're confused and looking at that line of the code specifically?

**Reply by alessandro-satanassi** [reply]
I understood that the function is used to tell the program that it is time to read the data from the cameras (by setting the variable "irPosUpdateTick = 1;"), but I did not understand what this line of code "pwm_hw->intr = 0xff;" does .
I'm asking this because I'm trying to port the firmware for the micro ESP32. Thank you

**Reply by SeongGino** [reply]
You'll want to ask @Prow7 since he's the one that wrote the timer irq blocks for previous boards, RP2040 included.

**Reply by Prow7** [reply]
That line clears the PWM IRQ flags.

**Reply by alessandro-satanassi** [reply]
Why were PWMs used and not Timers?

**Reply by SeongGino** [reply]
Because PWM is the most appropriate for a constant entity like the camera polling interval.

Anyways, I believe that should adequately answer the topic.

**Reply by Prow7** [reply]
The RP2040 doesn't have general purpose timers. You use a PWM channel with no output as a timer.

**Reply by alessandro-satanassi** [reply]
OK thank you. Now I get it. For the port to the Esp32 micro I will use timers instead (must generalize "IRCamUpdateRate = 209" interrupt per second)
